src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```
@@ -0,0 +1,395 @@
+        boolean hasPassword;
+        String meetingNumber;
+        String meetingPassword;
+
+        Pattern dialStringPattern = Pattern.compile("(\\d+)\\.?(\\d+)?(@zoomcrc\\.com)");
+        Matcher matcher = dialStringPattern.matcher(value);
+        if(!matcher.find() && !matcher.matches()){
+            // This might be the section to call out SIP/h323 devices it "invite" functionality fits our needs
+            if(logger.isErrorEnabled()) {
+                logger.error("Dial string does not match expected pattern meetingNumber.meetingPassword@zoomcrc.com. Skipping.");
+            }
+            return "";
+        } else {
+            hasPassword = !StringUtils.isEmpty(matcher.group(2));
+            meetingNumber = matcher.group(1);
+            meetingPassword = hasPassword ? matcher.group(2) : "";
+        }
+
+        if(logger.isDebugEnabled()){
+            logger.debug(String.format("Attempt to join or start a meeting with meetingNumber: %s and password: %s", meetingNumber, meetingPassword));
+        }
+
+        if(executeAndVerify(String.format(ZCOMMAND_DIAL_JOIN, meetingNumber) + (hasPassword ? (" password:" + meetingPassword) : "") + "\r", "*r DialJoinResult (status=OK)")){
+            isHost = false;
+            if(logger.isDebugEnabled()){
+                logger.debug("Joining the meeting: " + meetingNumber);
+            }
+            this.meetingNumber = meetingNumber;
+            return meetingNumber;
+        } else {
+            if(executeAndVerify(String.format(ZCOMMAND_DIAL_START, meetingNumber) + (hasPassword ? (" password:" + meetingPassword) : "") + "\r", "*r DialJoinResult (status=OK)")){
+                isHost = true;
+                if(logger.isDebugEnabled()){
+                    logger.debug("Starting the meeting: " + meetingNumber);
+                }
+                this.meetingNumber = meetingNumber;
+                return meetingNumber;
+            }
+        }
+        return "";

```

What does that mean? How we get here? If it is error case should throw an exception.